src/bootstrap/builder.rs

```diff
@@ -903,17 +903,7 @@ impl<'a> Builder<'a> {
     }
 
     /// Creates a new standalone builder for use outside of the normal process
-    pub fn new_standalone(
-        build: &mut Build,
-        kind: Kind,
-        paths: Vec<PathBuf>,
-        stage: Option<u32>,
-    ) -> Builder<'_> {
-        // FIXME: don't mutate `build`
-        if let Some(stage) = stage {
-            build.config.stage = stage;
-        }
-
+    pub fn new_standalone(build: &Build, kind: Kind, paths: Vec<PathBuf>) -> Builder<'_> {
         Self::new_internal(build, kind, paths.to_owned())
```

Comment:
at point there's no reason to have a separate `standalone` function and you can just make `new_internal` public.

this function is going away altogether in https://github.com/rust-lang/rust/pull/113644/commits/79c670037617df68097d68d9f81708b1c96bab75 though, so maybe just drop this commit for now?